### PR TITLE
[DH-3] try using a test fork of the chp

### PR DESCRIPTION
**Pull request by shaneknapp**
@consideRatio pointed me to this...  there were a couple of memory leaks in `node-http-proxy` that haven't been rolled in to a release yet, so let's test this out and see if it helps.

https://github.com/jupyterhub/configurable-http-proxy/issues/434#issuecomment-1924261459
https://github.com/http-party/node-http-proxy/pull/1559
https://github.com/Jimbly/http-proxy-node16/commit/56283e33edfc7aad8c2605dd493da8a196b4371d
https://github.com/consideRatio/configurable-http-proxy/commits/main/

### CODE REVIEW

**Review by ryanlovett — APPROVED** [review]
This looks like valid chart yaml to me, based on https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/main/jupyterhub/templates/proxy/deployment.yaml and https://z2jh.jupyter.org/en/stable/resources/reference.html#proxy-chp-extracommandlineflags.

At first I thought the timeouts were 1000 days, but I see the units are milliseconds.

File: hub/values.yaml [hunk 1]
```diff
@@ -32,6 +32,20 @@ jupyterhub:
...
+      # https://github.com/Jimbly/http-proxy-node16/commit/56283e33edfc7aad8c2605dd493da8a196b4371d
+      # https://github.com/consideRatio/configurable-http-proxy/commits/main/
+      #
+      image:
+        tag: 4.6.1-fork
+      extraCommandLineFlags:
+        # set the timeout and proxyTimeout to 24 hours (arg is in milliseconds)
+        # https://github.com/http-party/node-http-proxy?tab=readme-ov-file#options
+        - "--timeout 86400000"
+        - "--proxyTimeout 86400000"
```
Comment (consideRatio) [inline hub/values.yaml]:
```suggestion
        - "--proxyTimeout=86400000"
```

File: hub/values.yaml [hunk 2]
```diff
@@ -32,6 +32,20 @@ jupyterhub:
...
+      # https://github.com/http-party/node-http-proxy/pull/1559
+      # https://github.com/Jimbly/http-proxy-node16/commit/56283e33edfc7aad8c2605dd493da8a196b4371d
+      # https://github.com/consideRatio/configurable-http-proxy/commits/main/
+      #
+      image:
+        tag: 4.6.1-fork
+      extraCommandLineFlags:
+        # set the timeout and proxyTimeout to 24 hours (arg is in milliseconds)
+        # https://github.com/http-party/node-http-proxy?tab=readme-ov-file#options
+        - "--timeout 86400000"
```
Comment (consideRatio) [inline hub/values.yaml]:
```suggestion
        - "--timeout=86400000"
```

File: hub/values.yaml (hunk 1 above)
Comment (consideRatio) [reply hub/values.yaml]: Without this i expect its like a string with a space in it will be considered an invalid flag

Comment (ryanlovett) [reply hub/values.yaml]: @consideRatio For what it is worth, an example in the docs has a space.

https://z2jh.jupyter.org/en/stable/resources/reference.html#proxy-chp-extracommandlineflags
```yaml
proxy:
  chp:
    extraCommandLineFlags:
      - "--auto-rewrite"
      - "--custom-header {{ .Values.myCustomStuff }}"
```

Comment (shaneknapp) [reply hub/values.yaml]: intersting.  i'll leave the `=` sign there.

**Comment by shaneknapp** [comment]
hmm, interesting.  i did a manual deployment to test this out on a staging hub and got the following error in the chp pod:  `error: unknown option '--proxyTimeout=86400000'`

`timeout` worked just fine...  i'll comment out `proxyTimeout` and re-test my deployment.

**Comment by shaneknapp** [comment]
ok, commenting out `proxyTimeout` let me deploy, and `timeout` is showing up in the pod config:

```
Containers:
  chp:
    Container ID:  containerd://2400011a0b0b7fca551735393d81b8b6b1efc60b2080e3061ce669c526844098
    Image:         quay.io/jupyterhub/configurable-http-proxy:4.6.1-fork
    Image ID:      quay.io/jupyterhub/configurable-http-proxy@sha256:55345bfa2f47f4d0c995e25d8cd14a09ce20b0aaeed65df648ec818441629c87
    Ports:         8000/TCP, 8001/TCP
    Host Ports:    0/TCP, 0/TCP
    Command:
      configurable-http-proxy
      --ip=
      --api-ip=
      --api-port=8001
      --default-target=http://hub:$(HUB_SERVICE_PORT)
      --error-target=http://hub:$(HUB_SERVICE_PORT)/hub/error
      --port=8000
      --timeout=86400000
```

**Comment by shaneknapp** [comment]
ah, it's `proxy-timeout`!  https://github.com/jupyterhub/configurable-http-proxy?tab=readme-ov-file#command-line-options

**Comment by shaneknapp** [comment]
there we go!

```
Containers:
  chp:
    Container ID:  containerd://99cab96888c3d6609d224e6b78bc9936156a6f98a12d08105f1d7f6396d224fb
    Image:         quay.io/jupyterhub/configurable-http-proxy:4.6.1-fork
    Image ID:      quay.io/jupyterhub/configurable-http-proxy@sha256:55345bfa2f47f4d0c995e25d8cd14a09ce20b0aaeed65df648ec818441629c87
    Ports:         8000/TCP, 8001/TCP
    Host Ports:    0/TCP, 0/TCP
    Command:
      configurable-http-proxy
      --ip=
      --api-ip=
      --api-port=8001
      --default-target=http://hub:$(HUB_SERVICE_PORT)
      --error-target=http://hub:$(HUB_SERVICE_PORT)/hub/error
      --port=8000
      --timeout=86400000
      --proxy-timeout=86400000
```

**Comment by shaneknapp** [comment]
@consideRatio fyi, this looks quite promising.  i've been running the fork without timeouts on our two biggest/most active hubs since friday last week and have only had one `chp` oomkill between then and now.

here's the memory usage over the past 24 hours on one:
image

i'm keeping a close eye on these two hubs for the rest of the week and will roll out to our remaining hubs on monday.

since our core node has plenty of available ram, i'm also bumping the `chp`'s memory limit from 1Gi to 1.5Gi.  this should give is a little bit more headroom and minimize (hopefully) and additional oomkills.

**Pull request merged**